brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformer.java

```diff
@@ -184,79 +190,118 @@ private static boolean isTosca(Map<?,?> obj) {
         return createApplicationSpec(application.getName(), dt, id);
     }
 
-    private String getParentId(NodeTemplate nodeTemplate) {
-        Requirement host = nodeTemplate.getRequirements() != null ? nodeTemplate.getRequirements().get("host") : null;
-        if (host != null) {
-            for (RelationshipTemplate r : nodeTemplate.getRelationships().values()) {
-                if (r.getRequirementName().equals("host")) {
-                    return r.getTarget();
+    public static class DependencyTree{
+
+        private Map<String, String> roots = MutableMap.of();
+        private Map<String, String> parents = MutableMap.of();
+        private Map<String, List<String>> children = MutableMap.of();
```

Comment:
Use a [MultiMap](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Multimap.html) for this field.